spec/features/users_can_view_an_activity_as_xml_spec.rb

```diff
@@ -1,363 +1,283 @@
 
-        it_behaves_like "valid activity XML"
+    it "does not include empty optional dates" do
+      visit organisation_activity_path(organisation, activity, format: :xml)
+      optional_start_date = xml.at("iati-activity/activity-date[@type = '2']")
+      optional_end_date = xml.at("iati-activity/activity-date[@type = '4']")
 
-        it "does not include collaboration_type field" do
-          visit organisation_activity_path(organisation, activity, format: :xml)
-          expect(xml).not_to have_selector("iati-activity/collaboration-type")
-        end
-      end
+      expect(optional_start_date).to be_nil
+      expect(optional_end_date).to be_nil
+    end
+  end
 
-      context "when the activity is a programme activity" do
-        let(:activity) { create(:programme_activity, :with_transparency_identifier, organisation: organisation, partner_organisation_identifier: "IND-ENT-IFIER") }
-        let(:xml) { Nokogiri::XML::Document.parse(page.body) }
+  context "when the activity has a collaboration_type" do
+    before { activity.update(collaboration_type: "1") }
 
-        it_behaves_like "valid activity XML"
+    it "contains the relevant collaboration_type code" do
+      visit organisation_activity_path(organisation, activity, format: :xml)
+      expect(xml.at("iati-activity/collaboration-type/@code").text).to eq "1"
+    end
+  end
 
-        it "includes the activity aims/objectives" do
-          visit organisation_activity_path(organisation, activity, format: :xml)
-          expect(xml).to have_selector("iati-activity/description", count: 2)
-          descriptions = xml.xpath("//iati-activity/description/@type").to_a
-          expect(descriptions.last.value).to eq("2")
-        end
-      end
+  context "when the activity has policy markers" do
+    before do
+      activity.update(
+        policy_marker_gender: "not_targeted",
+        policy_marker_biodiversity: "significant_objective",
+        policy_marker_disability: "principal_objective",
+        policy_marker_desertification: "principal_objective_and_in_support_of_an_action_programme",
+        policy_marker_nutrition: "not_assessed",
+        policy_marker_climate_change_adaptation: "not_assessed",
+        policy_marker_climate_change_mitigation: "not_assessed",
+        policy_marker_disaster_risk_reduction: "not_assessed"
+      )
+    end
 
-      context "when the activity is a project" do
-        let(:activity) { create(:project_activity) }
-        let(:fund) { create(:fund_activity) }
-        let(:programme) { create(:programme_activity) }
-        let(:xml) { Nokogiri::XML::Document.parse(page.body) }
+    it "includes all the policy markers with reportable values for IATI" do
+      visit organisation_activity_path(organisation, activity, format: :xml)
+      expect(xml).to have_selector("iati-activity/policy-marker/@code", count: 4) # We do not report 'not_assessed' values to IATI
```

Comment:
I wonder if the comment in this line belongs in the `it` description rather than a comment? Do you think that could be covered by `"It includes all IATI-assessed policy markers"`?

Reply:
Huh, and there's already a spec for `it "does not include the policy markers with a value of 'not assessed'"`.

Reply:
> 



> Do you think that's what "with reportable values for IATI" was intended to mean? ETA: and that it's not clear enough in its phrasing?

Yeah, I think so - agree with your edit, which is probably why the comment was added.

Interesting that there's already another test for it, maybe we can remove one?

Reply:
Eh, one spec is checking that the expected number of markers is present, but not the value of each specific marker, whereas the `not_assessed` spec is specifically checking for exclusion. Because `it "includes all the policy markers with reportable values for IATI"` only checks the total number of nodes, it could pass by accident.

Should we rework the first spec, and check for specific markers rather than the number of nodes? Removing one spec doesn't speed up all that much the test suite, but it *is* good to leave every spec in a better place than we found it...

Reply:
Yeah, checking the values for the markers feels safer to me, as you said, checking the number of nodes could pass by accident. Definitely a fan of tidying up whenever there's a smell!